File: src/Viasfora.LanguageServices/Viasfora.LanguageService.CSharp/ArgumentValidationTagger/ArgumentValidationTaggerProvider.cs
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.Text;
using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Text.Tagging;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;
using Winterdom.Viasfora.Contracts;
using Winterdom.Viasfora.Languages;
using Winterdom.Viasfora.LanguageService.Core.ArgumentValidationTagger;

namespace Winterdom.Viasfora.LanguageService.CSharp.ArgumentValidationTagger {

  [Export(typeof(IViewTaggerProvider))]
  [ContentType(ContentTypes.CSharp)]
  [TagType(typeof(ArgumentValidationTag))]
  public sealed class ArgumentValidationTaggerProvider : BaseArgumentValidationTaggerProvider {

    [ImportingConstructor]
    public ArgumentValidationTaggerProvider(
      IVsfTelemetry telemetry,
      IClassificationTypeRegistryService classificationRegistry,
      ILanguageFactory languageFactory,
      IVsfSettings settings
    ) : base(telemetry, classificationRegistry, languageFactory, settings) {
    }

    public override TextSpan IsArgumentValidationSpan(SyntaxNode node) {
      if ( !SyntaxHelper.IsAnyIfArgumentThrowSyntaxStatement(node) )
        return new TextSpan();

      return node.FullSpan;
    }
  }
}

File: src/Viasfora.LanguageServices/Viasfora.LanguageService.Basic/ArgumentValidationTagger/ArgumentValidationTaggerProvider.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.Text
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Text.Tagging
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition
Imports Winterdom.Viasfora.Contracts
Imports Winterdom.Viasfora.Languages
Imports Winterdom.Viasfora.LanguageService.Core.ArgumentValidationTagger

Namespace ArgumentValidationTagger

  <Export(GetType(IViewTaggerProvider))>
  <ContentType(ContentTypes.VB)>
  <TagType(GetType(ArgumentValidationTag))>
  Public Class ArgumentValidationTaggerProvider
    Inherits BaseArgumentValidationTaggerProvider

    <ImportingConstructor>
    Public Sub New(
      telemetry As IVsfTelemetry,
      classificationRegistry As IClassificationTypeRegistryService,
      languageFactory As ILanguageFactory,
      settings As IVsfSettings
    )
      MyBase.New(telemetry, classificationRegistry, languageFactory, settings)
    End Sub

    Public Overrides Function IsArgumentValidationSpan(node As SyntaxNode) As TextSpan
      If Not SyntaxHelper.IsAnyIfArgumentThrowSyntaxStatement(node) Then
        Return New TextSpan()
      End If

      Return node.FullSpan
    End Function
  End Class

End Namespace